### PR TITLE
Disptach event 'update copyright' when mapboxlayer set copyright

### PR DESCRIPTION
**Pull request by edellucien**
# How to
Dispatch 'change:copyright' event as it was in react-spatial, for the copyright component to react on it.

# Others

- [ ] It's not a hack or at least an unauthorized hack :).
- [ ] The images added are optimized.
- [ ] Everything in ticket description has been fixed.
- [ ] The author of the MR has made its own review before assigning the reviewer.
- [ ] The title means something for a human being.
- [ ] The title contains [WIP] if it's necessary.
- [ ] Labels applied. if it's a release? a hotfix?
- [ ] The new class' members & methods are well documented
- [ ] Tests added.

### CODE REVIEW

File: src/ol/layers/MapboxLayer.js [hunk 1]
```diff
@@ -249,6 +249,9 @@ export default class MapboxLayer extends Layer {
          * @type {string}
          */
         this.copyright = getCopyrightFromSources(this.mbMap);
+        this.dispatchEvent({
+          type: 'change:copyright',
+        });
```
Comment (oterral) [inline src/ol/layers/MapboxLayer.js:254]: Now I remember why putting copyright in the properties object was a good idea:
https://github.com/geops/mobility-toolbox-js/pull/14/files

Revert this change in mobility-toolbox

**Review by oterral — CHANGES_REQUESTED** [review]
See comment

File: src/ol/layers/MapboxLayer.js (hunk 1 above)
Comment (edellucien) [reply src/ol/layers/MapboxLayer.js:254]: using set/get method for object properties doesn't seems to be the correct way.
Then copyright property are not corectly set to the layer anymore

File: src/common/layers/Layer.test.js [hunk 2]
```diff
@@ -95,4 +95,16 @@ describe('Layer', () => {
...
     }).toThrow(Error);
   });
+
+  test('should set and get copyright property.', () => {
+    const layer = new Layer({
+      name: 'Layer',
+      olLayer,
+    });
+    expect(layer).toBeInstanceOf(Layer);
+    expect(layer.get('copyright')).toEqual(undefined);
```
Comment (oterral) [inline src/common/layers/Layer.test.js:105]:
```suggestion
    expect(layer.copyright).toEqual(undefined);
```

File: src/common/layers/Layer.test.js [hunk 3]
```diff
@@ -95,4 +95,16 @@ describe('Layer', () => {
...
+
+    layer.set('copyright', '&copy; OSM Contributors');
```
Comment (oterral) [inline src/common/layers/Layer.test.js:107]:
```suggestion
    layer.copyright = '&copy; OSM Contributors';
```

File: src/common/layers/Layer.test.js [hunk 4]
```diff
@@ -95,4 +95,16 @@ describe('Layer', () => {
...
+    expect(layer.get('copyright')).toEqual('&copy; OSM Contributors');
```
Comment (oterral) [inline src/common/layers/Layer.test.js:108]:
```suggestion
    expect(layer.copyright).toEqual('&copy; OSM Contributors');
```

**Review by oterral — CHANGES_REQUESTED** [review]
Test also the initlization of copyright.

**Comment by edellucien** [comment]
> Test also the initlization of copyright.

Test added
